wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -888,6 +888,17 @@ public WorkunitInfo compileWUFromECL(WorkunitInfo wu ) throws Exception
             createdWU.setCluster(wu.getCluster());
             submitWU(createdWU); // if no exception proceed
             this.monitorWUToCompletion(createdWU);
+
+            //exceptions, etc. aren't always included in the submit response; do another request to get all workunit info
+            WUInfo parameters=new WUInfo();
```

Comment:
That would be fantastic. Currently hipie has to catch exceptions returning from createAndRunWuFromEcl, check the message to see if it's a timeout, and then make a call to WUInfo, passing in the jobname (since no workunit ID is returned) to try and get the actual status of the workunit. Perhaps instead of throwing an error, on timeout, a WorkunitInfo object could be returned containing the wuid + the timeout error?